core/src/main/scala/schema/validator/TroySchemaValidator.scala

```diff
@@ -36,25 +35,27 @@ object TroySchemaValidator {
 
     implicit class ResultOps[T](result: Result[T]) {
 
-      def toEither: Either[Seq[Message], T] =
+      def toValidatedNel: ValidatedNel[SchemaError, T] =
         result match {
-          case V.Success(res, _) => res.asRight
-          case V.Error(es, _)    => es.asLeft
+          case V.Success(res, _) => Validated.valid(res)
+          case V.Error(es, _) =>
+            Validated.invalid(
+              NonEmptyList
+                .fromList(es.map(_.toSchemaValidatorError).toList)
+                .getOrElse(NonEmptyList(SchemaValidatorError("Unknown error"), Nil)))
```

Comment:
This is good, a possible alternative is a nested pattern match.

```scala
case V.Success(res, _) =>
  Validated.valid(res)
case V.Error(e :: es, _) =>
  Validated.invalid(NonEmptyList(e, es).map(_.toSchemaValidationError))
case V.Error(_, _) =>
  Validate.invalidNel(SchemaValidationError("Unkown error"))
```

Reply:
Nice, the problem is that the value on `Error` it's a `Seq` and I should use something like:
```scala
case V.Error(Seq(e, es @ _ *), _) =>
```

Reply:
Missed that, the original will probably be simpler.